src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,41 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static addAlive(animal) {
+    Animal.alive.push(animal);
+  }
+
+  static removeNotAlive(animal) {
+    Animal.alive = Animal.alive.filter(beast => beast !== animal);
+  }
+
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    this.hidden = false;
+
+    Animal.addAlive(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(victim) {
+    if (victim instanceof Herbivore && victim.hidden === false) {
```

Comment:
use logical NO `!` to check if animal is hidden
```suggestion
    if (victim instanceof Herbivore && !victim.hidden) {
```